internal/charmstore/store.go

```diff
@@ -259,6 +280,10 @@ func (s *Store) insertEntity(entity *mongodoc.Entity) (err error) {
 		readPerm = []string{params.Everyone, entity.User}
 		writePerm = []string{entity.User}
 	}
+	promulgated := mongodoc.False
```

Comment:
I don't think you need this variable - you should be able do:

    Promulgated: entity.PromulgatedURL != nil,

below.